### PR TITLE
Fix invalid reads reported by valgrind

### PR DESCRIPTION
**Pull request by riyazhakki**
#### Describe Bug or Feature

Below are the invalid reads reported on valgrind build.

==349== Invalid read of size 4
==349==    at 0x4642C6: reply_hook_bg (mom_hook_func.c:3447)
==349==    by 0x464487: mom_process_background_hooks (mom_hook_func.c:3554)
==349==    by 0x4BF158: dispatch_task (work_task.c:142)
==349==    by 0x4BF394: default_next_task (work_task.c:308)
==349==    by 0x43C321: main (mom_main.c:9803)
==349==  Address 0xb619f2c is 252 bytes inside a block of size 6,560 free'd
==349==    at 0x4C2ACBD: free (vg_replace_malloc.c:530)
==349==    by 0x43F767: job_purge (job_func.c:829)
==349==    by 0x4642C0: reply_hook_bg (mom_hook_func.c:3436)
==349==    by 0x464487: mom_process_background_hooks (mom_hook_func.c:3554)
==349==    by 0x4BF158: dispatch_task (work_task.c:142)
==349==    by 0x4BF394: default_next_task (work_task.c:308)
==349==    by 0x43C321: main (mom_main.c:9803)
==349==  Block was alloc'd at
==349==    at 0x4C29BC3: malloc (vg_replace_malloc.c:299)
==349==    by 0x43EE83: job_alloc (job_func.c:321)
==349==    by 0x442D64: req_quejob (req_quejob.c:579)
==349==    by 0x470ED4: process_IS_CMD (mom_server.c:479)
==349==    by 0x470ED4: is_request (mom_server.c:1159)
==349==    by 0x46CFF8: do_rpp (mom_main.c:6268)
==349==    by 0x46D055: rpp_request (mom_main.c:6307)
==349==    by 0x48E3DB: process_socket (net_server.c:504)
==349==    by 0x48E591: wait_request (net_server.c:620)
==349==    by 0x43C7D9: finish_loop (mom_main.c:6573)
==349==    by 0x43C7D9: main (mom_main.c:9794)

Invalid read of size 4
   at 0x18CAC5: req_quejob (req_quejob.c:559)
   by 0x18ABB8: dispatch_request (process_request.c:682)
   by 0x1D393C: process_IS_CMD (mom_server.c:479)
   by 0x1D4FFC: is_request (mom_server.c:1158)
   by 0x1CA994: do_rpp (mom_main.c:6279)
   by 0x1CAA12: rpp_request (mom_main.c:6318)
   by 0x2000E1: process_socket (net_server.c:512)
   by 0x20047E: wait_request (net_server.c:628)
   by 0x1CAED3: finish_loop (mom_main.c:6595)
   by 0x1D20E3: main (mom_main.c:9827)
Address 0x9665f90 is 608 bytes inside a block of size 2,264 free'd Block was alloc'd at
   at 0x4C30D3B: free (in /usr/lib/valgrind/vgpreload_memcheck-amd64-linux.so)
   by 0x18B863: free_br (process_request.c:1479)
   by 0x18C023: reply_send (reply_send.c:303)
   by 0x18C7BD: reply_jobid (reply_send.c:553)
   by 0x18CA65: req_quejob (req_quejob.c:553)
   by 0x18ABB8: dispatch_request (process_request.c:682)
   by 0x1D393C: process_IS_CMD (mom_server.c:479)
   by 0x1D4FFC: is_request (mom_server.c:1158)
   by 0x1CA994: do_rpp (mom_main.c:6279)
   by 0x1CAA12: rpp_request (mom_main.c:6318)
   by 0x2000E1: process_socket (net_server.c:512)
   by 0x20047E: wait_request (net_server.c:628)

#### Describe Your Change

Fix the incorrect memory access code.

#### Attach Test and Valgrind Logs/Output

[valgrind.log.txt](https://github.com/PBSPro/pbspro/files/3903751/valgrind.log.txt)

### CODE REVIEW

**Comment by riyazhakki** [comment]
Testing on Cray

**Comment by riyazhakki** [comment]
[valgrind.log.txt](https://github.com/PBSPro/pbspro/files/3911457/valgrind.log.txt)

File: src/resmom/requests.c [hunk 1]
```diff
@@ -5029,6 +5029,18 @@ req_del_hookfile(struct batch_request *preq) /* ptr to the decoded request   */
...
+				/** 
+				 * This event runs hook in the background,
+				 * and it's deferred task created while
+				 * running the hook, is required for graceful
+				 * exit of the job. 
+				 */
+				reply_ack(preq);
+				return;
+			}
+#endif
```
Comment (bhroam) [inline src/resmom/requests.c]: We just removed all the win32 code.  Please remove this.

Comment (subhasisb) [reply src/resmom/requests.c]: Hi @bhroam this is mom code and we still have a mom in windows - we have only removed WIN32 code from server, sched and comm

**Pull request merged**